### PR TITLE
Added an Ingress resource for traffic routing

### PR DESCRIPTION
**Pull request by imzainhafeez**
- Added an Ingress resource to route traffic to the MongoDB and MongoDB Express services.
- Configured paths /mongo and /mongo-express to route to mongodb-service and mongo-express-service respectively.

### CODE REVIEW

File: k8s/ingress.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            port:
+              number: 8081
+      - path: /mongo
+        pathType: Prefix
+        backend:
+          service:
+            name: mongodb-service
+            port:
+              number: 27017
+
```
Comment (Abdullahxz) [inline k8s/ingress.yml]: Please remove this extra line at the end of file.

File: k8s/ingress.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            name: mongo-express-service
+            port:
+              number: 8081
+      - path: /mongo
+        pathType: Prefix
+        backend:
+          service:
+            name: mongodb-service
+            port:
+              number: 27017
```
Comment (Abdullahxz) [inline k8s/ingress.yml]: What's the hostname for this application?

**Review by Abdullahxz — CHANGES_REQUESTED** [review]

File: k8s/ingress.yml (hunk 1 above)
Comment (imzainhafeez) [reply k8s/ingress.yml]: done.

File: k8s/ingress.yml (hunk 2 above)
Comment (imzainhafeez) [reply k8s/ingress.yml]: No hostname specified for this.

Comment (Abdullahxz) [reply k8s/ingress.yml]: Please specify one.

**Review by Abdullahxz — CHANGES_REQUESTED** [review]
Please specify a hostname in ingress resource.

File: k8s/ingress.yml (hunk 2 above)
Comment (imzainhafeez) [reply k8s/ingress.yml]: Which hostname should be specified as I am using minikube and does not have hostname? Should i configure a dummy hostname inside /etc/hosts?

Comment (Abdullahxz) [reply k8s/ingress.yml]: Yes, please.

**Review by Abdullahxz — COMMENTED** [review]
Kindly use `/etc/hosts` and define a hostname for this ingress.

File: k8s/ingress.yml (hunk 2 above)
Comment (imzainhafeez) [reply k8s/ingress.yml]: okay

**Review by Abdullahxz — CHANGES_REQUESTED** [review]
It seems this ingress resource only exposes express and not the application. Please fix this. You can use path based routing to expose both under same hostname. Also, kindly use an appropriate hostname, such as www.myapp.com.

**Comment by imzainhafeez** [comment]
Done and with "www" as hostnames, the browsers are not allowing to open them with http, so used a more appropriate host but without www.

**Comment by Abdullahxz** [comment]
> Done and with "www" as hostnames, the browsers are not allowing to open them with http, so used a more appropriate host but without www.

Is there a reason behind this behaviour?

**Comment by imzainhafeez** [comment]
> > Done and with "www" as hostnames, the browsers are not allowing to open them with http, so used a more appropriate host but without www.
> 
> Is there a reason behind this behaviour?

My bad, no reason for this, behaviors are same for both dummy domain, actually the browser redirected to https at that time which gave that behavior, I missed that part.

**Review by Abdullahxz — APPROVED** [review]
LGTM, approved.

**Comment by Abdullahxz** [comment]
On second thought, How would you connect to the mongodb instance from outside the cluster?

**Comment by imzainhafeez** [comment]
> On second thought, How would you connect to the mongodb instance from outside the cluster?

through using nodeport

**Comment by Abdullahxz** [comment]
> > On second thought, How would you connect to the mongodb instance from outside the cluster?
> 
> through using nodeport

Are you sure? Why are you choosing nodeport over including that in the ingress? Also, is nodepoet highly available?

**Review by Abdullahxz — CHANGES_REQUESTED** [review]
> > On second thought, How would you connect to the mongodb instance from outside the cluster?
> 
> through using nodeport

Are you sure? Why are you choosing nodeport over including that in the ingress? Also, is nodepoet highly available?

**Comment by imzainhafeez** [comment]
> > > On second thought, How would you connect to the mongodb instance from outside the cluster?
> > 
> > 
> > through using nodeport
> 
> Are you sure? Why are you choosing nodeport over including that in the ingress? Also, is nodepoet highly available?

Umm, its one of the option, it was simple but yes it is not secure to expose node's port like that and yes it is not highly available as if a worker node fails the traffic coming to that one will not be served, we can use ingress for this, I did not completely know the downsides of using node port before this.

**Comment by imzainhafeez** [comment]
MongoDB service is now accessible through ingress.

File: k8s/ingress.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: mongo-ingress
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /
+spec:
+  rules:
+  - host: www.dogimages.com
```
Comment (Abdullahxz) [inline k8s/ingress.yml]: Using the same host for all three different services, even tho they are not logically related?

**Review by Abdullahxz — CHANGES_REQUESTED** [review]

File: k8s/ingress.yml (hunk 3 above)
Comment (imzainhafeez) [reply k8s/ingress.yml]: Added a different host for mongo-express and mongodb.

**Review by Abdullahxz — APPROVED** [review]
LGTM, approved.

**Pull request merged**